### PR TITLE
feat(web): use shortcut to favorite and archive on asset-grid

### PR DESCRIPTION
**Pull request by martabal**
Implements #9466

- With this PR, you can use shortcuts directly on the asset-grid to favorite and archive assets.
- Fixes an issue where, if you select multiple assets and press the "Favorite"/"Remove from favorites" button in the AssetSelectControlBar, there's no heart added/removed on the asset thumbnails

### CODE REVIEW

File: web/src/lib/components/photos-page/asset-grid.svelte [hunk 1]
```diff
@@ -128,6 +146,14 @@
         { shortcut: { key: 'D', ctrl: true }, onShortcut: () => deselectAllAssets() },
         { shortcut: { key: 's' }, onShortcut: () => onStackAssets() },
       );
+      // you're not supposed to manage your assets on the trash page: there's no button to manage it, then there's no reason to have shortcuts to do it.
+      if ($page.url.pathname !== AppRoute.TRASH) {
+        shortcuts.push({ shortcut: { key: 'a', shift: true }, onShortcut: onArchive });
+        // on some page, you can't favorite assets (like on `/parthner`)
```
Comment (danieldietzler) [inline web/src/lib/components/photos-page/asset-grid.svelte:152]:
```suggestion
        // on some pages, you can't favorite assets (like on `/partner`)
```

Comment (danieldietzler) [reply web/src/lib/components/photos-page/asset-grid.svelte:152]: Where do you enforce this?

Comment (martabal) [reply web/src/lib/components/photos-page/asset-grid.svelte]: With `isAllFavorite` != `null`

Comment (danieldietzler) [reply web/src/lib/components/photos-page/asset-grid.svelte]: Should that be called `isAllowFavorite` then? I assumed `isAllFavorite` would be true if all selected assets are already favorites

Comment (martabal) [reply web/src/lib/components/photos-page/asset-grid.svelte]: This variable allows us to know two things: 
- know if the assets should be liked or unliked. 
- to find out if the assets can be liked

Comment (danieldietzler) [reply web/src/lib/components/photos-page/asset-grid.svelte]: Ah, so `isAllFavorite === true` means that they should be unliked (`=== false` means they should be liked), and if it's null they can't be liked?

Comment (martabal) [reply web/src/lib/components/photos-page/asset-grid.svelte]: Yes, exactly

File: web/src/routes/(user)/people/+page.svelte [hunk 2]
```diff
@@ -463,7 +461,7 @@
             name="name"
             type="text"
             bind:value={personName}
-            bind:this={changeNameInputEl}
```
Comment (zackpollard) [inline web/src/routes/(user)/people/+page.svelte:466]: This seems unrelated to this change and imo is not something we should encourage by making generic. I'm not sure why we need an arbitrary 100ms delay here, but there's likely something else going on that needs to be resolved.

Comment (martabal) [reply web/src/routes/(user)/people/+page.svelte:466]: IIRC, it's because the FullScreenModal has a transition delay. 

I didn't want to open a PR for such a small change, but if you prefer I can open a new one.

File: web/src/lib/utils/actions.ts [hunk 3]
```diff
@@ -1,11 +1,11 @@
 import { notificationController, NotificationType } from '$lib/components/shared-components/notification/notification';
-import { deleteAssets as deleteBulk, type AssetResponseDto } from '@immich/sdk';
+import { deleteAssets as deleteBulk, updateAssets, type AssetResponseDto } from '@immich/sdk';
 import { handleError } from './handle-error';
 
 export type OnDelete = (assetIds: string[]) => void;
 export type OnRestore = (ids: string[]) => void;
 export type OnArchive = (ids: string[], isArchived: boolean) => void;
-export type OnFavorite = (ids: string[], favorite: boolean) => void;
+export type OnFavorite = (assets: AssetResponseDto[], isFavorite: boolean) => void;
```
Comment (alextran1502) [inline web/src/lib/utils/actions.ts:8]: Why do we need to pass the whole AssetResponseDto here?

File: web/src/lib/components/photos-page/asset-grid.svelte [hunk 4]
```diff
@@ -34,6 +35,7 @@
   export let isShared = false;
   export let album: AlbumResponseDto | null = null;
   export let isShowDeleteConfirmation = false;
+  export let isAllFavorite: boolean | null = null;
```
Comment (alextran1502) [inline web/src/lib/components/photos-page/asset-grid.svelte:38]: Can you explain the use case of the variable a bit?

Comment (martabal) [reply web/src/lib/components/photos-page/asset-grid.svelte:38]: Of course!

The idea is to have only one reactive declaration to know if the selected assets need to be favored or unfavoured. 

But currently, not all assets can be favored, so if `isAllFavorite` is null, we don't allow that shortcut on that asset-grid view.
